BrightID/src/components/Apps/appThunks.ts

```diff
@@ -0,0 +1,588 @@
+    const linkingError = selectLinkingAppError(getState());
+    if (linkingError) {
+      console.log(
+        `Can't request linking when there is still an active error. Current errir: ${linkingError}`,
+      );
+      return;
+    }
+
+    // store app linking details
+    dispatch(setLinkingAppInfo(params));
+
+    const api = getGlobalNodeApi();
+    dispatch(setAppLinkingStep({ step: app_linking_steps.REFRESHING_APPS }));
+    try {
+      // make sure to have latest appInfo available
+      const apps = await api.getApps();
+      dispatch(setApps(apps));
+    } catch (e) {
+      const msg = e instanceof Error ? e.message : `${e}`;
+      dispatch(setLinkingAppError(`Failed to fetch latest appInfo: ${msg}`));
+      return;
+    }
+
+    // First check if provided data is valid
+    const appInfo = selectAppInfoByAppId(getState(), params.appId);
+    if (!appInfo) {
+      // The app that should be linked was not found!
+      dispatch(
+        setLinkingAppError(
+          t('apps.alert.text.invalidContext', {
+            context: `${params.appId}`,
+          }),
+        ),
+      );
+      return;
+    }
+
+    if (params.v === 6 && !appInfo.usingBlindSig) {
+      // v6 apps HAVE to use blind sigs!
+      dispatch(
+        setLinkingAppError(
+          t('apps.alert.text.invalidApp', {
+            app: `${params.appId}`,
+          }),
+        ),
+      );
+      return;
+    }
+
+    // request user confirmation
+    dispatch(
+      setAppLinkingStep({ step: app_linking_steps.WAITING_USER_CONFIRMATION }),
+    );
+  };
+
+export const startLinking =
+  (): AppThunk<Promise<void>> => async (dispatch: AppDispatch, getState) => {
+    const applinkingStep = selectApplinkingStep(getState());
+    if (applinkingStep !== app_linking_steps.WAITING_USER_CONFIRMATION) {
+      console.log(
+        `Can't start linkApp when not in WAITING_USER_CONFIRMATION state. Current state: ${applinkingStep}`,
+      );
+    }
+    const isSponsored = selectIsSponsored(getState());
+    if (!isSponsored) {
+      // trigger sponsoring workflow
+      console.log(`Not yet sponsored, proceed with sponsoring`);
+      const opHash = await dispatch(requestSponsoring());
+      if (opHash) {
+        console.log(`Sponsor op hash: ${opHash}`);
+        dispatch(setSponsorOperationHash(opHash));
+      }
+    } else {
+      // trigger app linking
+      console.log(`Already sponsored, proceed with linking`);
+      dispatch(setAppLinkingStep({ step: app_linking_steps.SPONSOR_SUCCESS }));
+      dispatch(linkAppOrContext());
+    }
+  };
+
+export const linkAppOrContext =
+  (): AppThunk<Promise<void>> => async (dispatch: AppDispatch, getState) => {
+    const applinkingStep = selectApplinkingStep(getState());
+    if (applinkingStep !== app_linking_steps.SPONSOR_SUCCESS) {
+      console.log(
+        `Can't start linkApp when not in SUCCESS state. Current state: ${applinkingStep}`,
+      );
+    }
+    // TODO How to handle below condition?
```

Comment:
In the previous version, a recursive call and a sleep were used to continue linking after blind sigs were updated.
It seems that is removed, and this TODO is missed to be implemented.

Reply:
This should be handled properly now. If blind sigs are currently being updated, the app will wait up to 2 minutes for that update to finish. If the update does not finish in time an error will be shown and app linking aborted.